auth/strategies.js

```diff
@@ -0,0 +1,55 @@
+const passport = require('passport');
+const {BasicStrategy} = require('passport-http');
+const {Strategy: JwtStrategy, ExtractJwt} = require('passport-jwt');
+
+const {User} = require('../users/models');
+
+const basicStrategy = new BasicStrategy((username, password, callback) => {
+  let user;
+  User
+    .findOne({username: username})
+    .then(_user => {
+      user = _user;
+      if (!user) {
+        return Promise.reject({
```

Comment:
Probably should add a comment here explaining why we're rejecting promise (aka, so further up stack we raise an error...)